zmon-controller-app/src/main/java/org/zalando/zmon/api/VisualizationApi.java

```diff
@@ -0,0 +1,84 @@
+package org.zalando.zmon.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.zalando.zmon.security.permission.DefaultZMonPermissionService;
+import org.zalando.zmon.service.VisualizationService;
+
+import java.util.Map;
+
+@Controller
+@RequestMapping(value = "/api/v1/visualization")
+public class VisualizationApi {
+
+    private VisualizationService visualizationService;
+    private DefaultZMonPermissionService authService;
+    private final Logger log = LoggerFactory.getLogger(VisualizationApi.class);
+
+
+    @Autowired
+    public VisualizationApi(VisualizationService visualizationService,
+                            DefaultZMonPermissionService authService) {
+        this.visualizationService = visualizationService;
+        this.authService = authService;
+    }
+
+    @RequestMapping(value = "/")
+    public String visualizationHomeRedirect() {
+        return visualizationService.homeRedirect();
+    }
+
+    @RequestMapping(value = "/script")
+    public String grafanaScriptedDashboardRedirect(@RequestParam Map<String, String> params) {
+        return visualizationService.dynamicDashboardRedirect(params);
+    }
+
+    @ResponseBody
+    @RequestMapping(value = "/dashboards/{id}", method = RequestMethod.GET)
+    public ResponseEntity<JsonNode> getDashboard(@PathVariable(value = "id") String id, @RequestHeader("Authorization") String authHeader) {
+        if (!authService.hasUserAuthority()) {
```

Comment:
It's usually a bad practice to check auth inside controller's methods, as it have to be taken care by Spring Security. Are you sure there's no other way?